src/Uno.UI/UI/Xaml/UIElement.skia.cs

```diff
@@ -215,7 +216,26 @@ private void InnerRemoveChild(UIElement child)
 
 		internal UIElement FindFirstChild() => _children.FirstOrDefault();
 
-		public string Name { get; set; }
+		#region Name Dependency Property
+
+		private void OnNameChanged(string oldValue, string newValue)
+		{
+			if (FrameworkElementHelper.IsUiAutomationMappingEnabled)
+			{
+				Windows.UI.Xaml.Automation.AutomationProperties.SetAutomationId(this, newValue);
+			}
+		}
+
+		[GeneratedDependencyProperty(DefaultValue = "", ChangedCallback = true)]
+		internal static DependencyProperty NameProperty { get; } = CreateNameProperty();
+
+		public string Name
+		{
+			get => GetNameValue();
+			set => SetNameValue(value);
+		}
+
```

Comment:
Why are we adding the property in UIElement? Shouldn't it be in FrameworkElement only?

Reply:
I put it on `UIElement` as `Name` property is also on `UIElement` on Skia and WASM currently - so to avoid breaking change and also be able to use generated dependency property, I put it on `UIElement` level for now as `internal` and as part of breaking changes it can than be moved directly to `FrameworkElement` along with the `Name` property. I have already added it to the breaking changes rollup list 😀 .

Reply:
@MartinZikmund Can't we keep the public `Name` property in UIElement with `base.Name` calls, and have the `NameProperty` along with a `Name` property in `FrameworkElement`?

That would avoid the breaking change, and also not introducing `NameProperty` in UIElement.

Not sure if I'm missing something?

Reply:
The one disadvantage there is that if someone, for some reason, used the `UIElement.Name` property directly, its value would not be reflected in `FrameworkElement.NameProperty` if done this way (unless the setter of `UIElement.Name` would also set the dependency property, but that sounds a bit convoluted 🤔)